### PR TITLE
Fix startup skills TOC and semantic lookup

### PR DESCRIPTION
**Pull request by abrookins**
## Summary
- build the startup skills TOC from an unfiltered `skills_check` call instead of reusing the user query
- cap the startup skills TOC with `startup_skills_toc_limit`, now defaulting to `50`
- add truncation guidance so the startup context tells the model to search for related skills when more are available than shown
- rank queried skills semantically in the AFS workspace backend for both API and gateway-backed paths
- cover the startup call shape, truncation behavior, and semantic ranking behavior with unit tests

## Verification
- `./.venv/bin/ruff check redis_sre_agent/agent/knowledge_context.py redis_sre_agent/core/config.py redis_sre_agent/skills/afs_workspace_backend.py tests/unit/agent/test_knowledge_context.py tests/unit/skills/test_afs_workspace_backend.py`
- `./.venv/bin/pytest -q tests/unit/agent/test_knowledge_context.py tests/unit/skills/test_afs_workspace_backend.py`
- `./.venv/bin/pytest -q tests/unit/agent/test_knowledge_context.py tests/unit/evaluation/test_injection.py tests/unit/evaluation/test_prompt_scenarios.py`
- local prompt QA confirmed the startup context emits: `50 skills shown, 23 more available` and instructs the model to search for related skills when the catalog is truncated
- live QA on runtime deployment `dep_20260508_232733` / version `ver_20260508_232733`
  - startup trace showed `skills_check(query="", limit=50, offset=0, version="latest")`
  - `run a health check` retrieved and used `redis-cluster-health-check`
  - semantic queries like `health check skill` and `run a health check` ranked `redis-cluster-health-check` first

---

> [!NOTE]
> **Medium Risk**
> Changes how startup grounding context is constructed and how skills are searched/ranked, which can affect agent behavior and prompt content across chat/knowledge/langgraph flows. Adds semantic embedding-based ranking and larger catalog fetches in the skills backend, introducing performance and dependency risk around the vectorizer.
> 
> **Overview**
> Startup grounding no longer derives the skills TOC from the user query; `build_startup_knowledge_context` now builds an *inventory-style* skills list using `skills_query` (default `None`) and a new `settings.startup_skills_toc_limit` (default `50`), and adds explicit truncation guidance when more skills exist than are shown.
> 
> All agents (`chat_agent`, `knowledge_agent`, `langgraph_agent`) stop passing `query` into startup context building, and internal tool envelopes/tests are updated to match the new call shape and limits.
> 
> `AFSWorkspaceSkillBackend.list_skills` now semantically ranks queried skill results (API and gateway paths) using embeddings + a small lexical overlap term, fetching up to 500 catalog entries when needed; new unit tests cover truncation messaging, default limit behavior, semantic ranking, and API param handling (including omitting `version` when `None`).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 092e8a9e4fc9893586df56ab82af3ecd542ad7ac. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: redis_sre_agent/agent/knowledge_context.py [hunk 1]
```diff
@@ -254,14 +255,19 @@ async def build_startup_knowledge_context(
     version: Optional[str] = "latest",
     pinned_limit: int = 20,
     pinned_content_char_budget: int = 12000,
-    skills_limit: int = 20,
```
[the remark this answers is not in the document]
Comment (abrookins) [reply redis_sre_agent/agent/knowledge_context.py:257]: Fixed in `a79c06b`.

I removed the dead `query` parameter from `build_startup_knowledge_context` and updated the agent and evaluation call sites so the startup inventory path is explicitly queryless instead of carrying a required no-op argument.

File: redis_sre_agent/skills/afs_workspace_backend.py [hunk 2]
```diff
@@ -260,61 +262,46 @@ async def _list_skills_via_api(
...
+                "limit": limit,
+                "offset": offset,
+            },
         )
-        if query:
-            matches = data.get("matches", [])
-            if isinstance(matches, list):
-                by_identity = {
-                    (
-                        str(item.get("skillSlug", "")).strip(),
```
[the remark this answers is not in the document]
Comment (abrookins) [reply redis_sre_agent/skills/afs_workspace_backend.py:295]: Fixed in `a79c06b`.

The API-backed list helpers now add `version` only when it is truthy, so the httpx path no longer serializes `None` as `?version=`. I also added a regression test that asserts the request params omit `version` when `version=None`.

File: redis_sre_agent/agent/knowledge_context.py [hunk 3]
```diff
@@ -61,6 +67,14 @@ def _skills_toc_lines(skills: List[Dict[str, Any]]) -> List[str]:
         "- If a listed skill matches the request, retrieve it with `get_skill` before claiming that you used, followed, or executed it.",
         "- If you use any retrieved skills during your turn, add a note to your answer saying which skill(s) you used.",
     ]
+    if total_skills is not None and total_skills > len(skills):
+        remaining = total_skills - len(skills)
+        shown = displayed_limit if displayed_limit is not None else len(skills)
```
[the remark this answers is not in the document]
Comment (abrookins) [reply redis_sre_agent/agent/knowledge_context.py]: Fixed in `ca2bda1`.

`_skills_toc_lines` now reports the actual number of returned skills with `len(skills)` instead of the requested display limit, and I added a regression test for the case where the backend returns fewer rows than requested.

File: redis_sre_agent/agent/knowledge_context.py [hunk 4]
```diff
@@ -49,7 +50,12 @@ def merge_internal_tool_envelopes(
     return merged
 
 
-def _skills_toc_lines(skills: List[Dict[str, Any]]) -> List[str]:
+def _skills_toc_lines(
+    skills: List[Dict[str, Any]],
+    *,
+    total_skills: Optional[int] = None,
+    displayed_limit: Optional[int] = None,
```
[the remark this answers is not in the document]
Comment (abrookins) [reply redis_sre_agent/agent/knowledge_context.py]: Fixed in `6987c23`.

I removed the dead `displayed_limit` keyword from `_skills_toc_lines` and dropped the matching call-site argument in `build_startup_knowledge_context`, since the helper now derives the shown count directly from `len(skills)`.
